From 99d3ceb650d4a05fdc26987e3dd436dccf3206bb Mon Sep 17 00:00:00 2001
From: Linn Crosetto <linn@hpe.com>
Date: Fri, 1 Apr 2016 10:44:15 -0600
Subject: [PATCH] Add support for UEFI Secure Boot to qemu-efi

Add SECURE_BOOT_ENABLE flag to aarch64 build to enable support for UEFI
Secure Boot.

Signed-off-by: Linn Crosetto <linn@hpe.com>
---
 debian/rules | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/debian/rules b/debian/rules
index 13560c2..54e22e1 100755
--- a/debian/rules
+++ b/debian/rules
@@ -49,11 +49,11 @@ setup-build:
 	# single shell invocation
 	. ./edksetup.sh
 	QUILT_PC=.pc-post QUILT_PATCHES=debian/post-patches quilt push -a || [ $$? = 2 ]
+	cd CryptoPkg/Library/OpensslLib/ && ./Install.sh
 
 build-ovmf: EDK2_ARCH_DIR=X64
 build-ovmf:
 ifneq (,$(findstring ovmf, $(shell dh_listpackages)))
-	cd CryptoPkg/Library/OpensslLib/ && ./Install.sh
 	cd UefiCpuPkg/ResetVector/Vtf0 && python Build.py
 	mkdir -p EdkShellBinPkg/FullShell/$(EDK2_ARCH_DIR) \
 	         FatBinPkg/EnhancedFatDxe/$(EDK2_ARCH_DIR)
@@ -86,6 +86,7 @@ build-qemu-efi:
 		GCC49_AARCH64_PREFIX=$(GCC49_AARCH64_PREFIX) build -a $(EDK2_HOST_ARCH) \
 			-t $(EDK2_TOOLCHAIN) \
 			-p ArmVirtPkg/ArmVirtQemu.dsc \
+			-DSECURE_BOOT_ENABLE=TRUE \
 			-DINTEL_BDS \
 			-b RELEASE
 
-- 
2.8.0.rc3

